### PR TITLE
Feature/evidence

### PR DESCRIPTION
**Pull request by aapomm**
### Spec
The `issueid` attribute that comes from the Veracode file is not unique. This isn't an ideal use for the plugin_id since it will deduplicate distinct vulnerabilities.

**Proposed solution**
- Use `cweid` as the plugin_id instead of the current issueid.
- Also, implement Evidence so that Description and other fields can be different from one instance of a flaw to the next.

### Testing Steps

Provide steps to test functionality, described in detail for someone not familiar with this part of the application / code base

### Other Information

If there's anything else that's important and relevant to your pull
request, mention that information here. This could include
benchmarks, or other information.

Thanks for contributing to Dradis!

### Copyright assignment

Collaboration is difficult with commercial closed source but we want
to keep as much of the OSS ethos as possible available to users
who want to fix it themselves.

In order to unambiguously own and sell Dradis Framework commercial
products, we must have the copyright associated with the entire
codebase. Any code you create which is merged must be owned by us.
That's not us trying to be a jerks, that's just the way it works.

You can delete this section, but the following sentence needs to
remain in the PR's description:

> I assign all rights, including copyright, to any future Dradis
> work by myself to Security Roots.

### Check List

- [x] Added a CHANGELOG entry
- [x] Added specs

### CODE REVIEW

File: lib/dradis/plugins/veracode/importer.rb [hunk 1]
```diff
@@ -52,15 +52,21 @@ def parse_report_details(xml_detailedreport)
...
     end
 
-
     def parse_flaw(xml_flaw)
-      flaw_id = xml_flaw[:issueid]
-      logger.info{ "\t\t => Creating new issue (flaw issueid: #{ flaw_id })" }
+      cwe_id = xml_flaw[:cweid]
+      logger.info { "\t\t => Creating issue and evidence (flaw cweid: #{ cwe_id })" }
+
+      logger.info { "Adding report details (app_name: #{ app_name })" }
```
Comment (caitmich) [inline lib/dradis/plugins/veracode/importer.rb]: You need an accessor for `app_name` to reference it here

File: lib/dradis/plugins/veracode/importer.rb [hunk 2]
```diff
@@ -52,15 +52,21 @@ def parse_report_details(xml_detailedreport)
...
+      flaw = ::Veracode::Flaw.new(xml_flaw)
+      issue_text = template_service.process_template(template: 'issue', data: flaw)
+      issue = content_service.create_issue(text: issue_text, id: cwe_id)
 
-      issue_text = template_service.process_template(template: 'issue', data: xml_flaw)
-      issue = content_service.create_issue(text: issue_text, id: flaw_id)
+      veracode_evidence = ::Veracode::Evidence.new(xml_flaw)
```
Comment (caitmich) [inline lib/dradis/plugins/veracode/importer.rb:67]: is veracode necessary here? We're just using `flaw` above

File: lib/dradis/plugins/veracode/field_processor.rb [hunk 3]
```diff
@@ -2,25 +2,30 @@ module Dradis
   module Plugins
     module Veracode
       class FieldProcessor < Dradis::Plugins::Upload::FieldProcessor
-        def post_initialize(args={})
+        def post_initialize(args = {})
```
Comment (caitmich) [inline lib/dradis/plugins/veracode/field_processor.rb:5]: `attr_accessor :record` ?

File: lib/dradis/plugins/veracode/importer.rb (hunk 1 above)
Comment (aapomm) [reply lib/dradis/plugins/veracode/importer.rb]: Removed

File: lib/dradis/plugins/veracode/field_processor.rb (hunk 3 above)
Comment (aapomm) [reply lib/dradis/plugins/veracode/field_processor.rb:5]: I don't think we need this? The other field processors don't have accessors too

File: lib/dradis/plugins/veracode/importer.rb [hunk 4]
```diff
@@ -25,11 +25,11 @@ def import(params={})
       end
 
       # create app_name, and parse attributes
-      parse_report_details(xml.root)
+      @node = parse_report_details(xml.root)
```
Comment (etdsoft) [inline lib/dradis/plugins/veracode/importer.rb]: can this be an argument we pass to #parse_flaw instead? Doesn't seem like a useful attribute (or related to the Importer) class.

**Pull request merged**